src/app/shared/code-editor/code-editor.component.ts

```diff
@@ -57,13 +59,43 @@ export class CodeEditorComponent implements AfterViewInit {
       fontSize: '12pt',
     });
 
-    this.editor.getSession().on('change', () => {
-      this.contentChange.emit(this.editor.getValue());
-    });
-
     // Set content if possible
-    if (this.editorContent) {
-      this.editor.setValue(this.editorContent, -1);
+    const setContent = (content: string, cursorPos = 0): void => {
+      this.editorContent = content;
+      if (this.editorContent) {
+        this.editor.setValue(this.editorContent, cursorPos);
+      }
+
+      this.editor.getSession().on('change', () => {
+        this.contentChange.emit(this.editor.getValue());
+      });
+
+      this.editor.focus();
+    };
+
+    let sampleCodeUrl = '';
+
+    // Load helloworld files by default when editing empty CWL/WDL/NF files
+    if (!this.editorContent) {
+      if (this.mode === 'cwl') {
+        if (this.entryType === 'tool') {
+          sampleCodeUrl = 'assets/text/helloworld-tool.cwl';
+        } else if (this.entryType === 'workflow') {
+          sampleCodeUrl = 'assets/text/helloworld-workflow.cwl';
+        }
+      } else if (this.mode === 'wdl') {
+        sampleCodeUrl = 'assets/text/helloworld.wdl';
+      } else if (this.mode === 'nfl') {
+        sampleCodeUrl = 'assets/text/helloworld.nf';
+      } else if (this.editorFilepath === '/nextflow.config') {
+        sampleCodeUrl = 'assets/text/nextflow.config';
+      }
+    }
+    if (sampleCodeUrl) {
+      const httpOptions: Object = { responseType: 'text' };
+      this.httpClient.get(sampleCodeUrl, httpOptions).subscribe(setContent);
```

Comment:
It's unfortunate that it's another http request, but there's always a tradeoff. @garyluu any strong opinions on this? These files are so small it probably doesn't matter either way, but wondering if this best practice or not.

Reply:
I don't have a better solution that this right now.

Reply:
Yeah, in my mind it's either this or add bloat to the bundle.